### PR TITLE
AGENT-161: install-config validations for agent installer

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
To be able to validate the install-config.yaml as per agent installer requirements, 
the install-config is first loaded early in the agent installer OptionalInstallConfig.
Then, the following validations are performed if the loaded file is valid YAML.

- Only Baremetal, VSphere, and None platforms are supported
e.g. `invalid install-config configuration: Platform: Unsupported value: "aws": supported values: "baremetal", "vsphere", "none"`
- API and Ingress VIPs must be set when using Baremetal and VSphere platform
e.g. 
 1. `invalid install-config configuration: Platform.Baremetal.ApiVip: Required value: ApiVip must be set for baremetal platform`
 2. `invalid install-config configuration: Platform.VSphere.IngressVip: Required value: IngressVip must be set for vsphere platform`

- If using none platform, the number of control plane replicas should be 1 and 
the total number of worker replicas should be 0
e.g. 
`invalid install-config configuration: [ControlPlane.Replicas: Required value: control plane replicas must be 1 for none platform. Found 3, Compute.Replicas: Required value: total number of worker replicas must be 0 for none platform. Found 2]`

Once these agent installer specific validations are passed, we rely on the usual generic validations offered by installer repo for `pkg/asset/installconfig/installconfig.go`

### CODE REVIEW

File: pkg/asset/agent/installconfig_test.go
Comment (bfournie) [inline pkg/asset/agent/installconfig_test.go:172]: Could use constant for install-config.yaml here

File: pkg/asset/agent/installconfig.go
Comment (bfournie) [inline pkg/asset/agent/installconfig.go:97]: Nit - would be good to use same definition of allErrs as the other functions, i.e.g:
`var allErrs field.ErrorList`

File: pkg/asset/agent/installconfig.go
Comment (bfournie) [inline pkg/asset/agent/installconfig.go:159]: Just a question - does platform None always imply SNO?

Comment (pawanpinjarkar) [reply pkg/asset/agent/installconfig.go:159]: Yes

File: pkg/asset/agent/installconfig.go
Comment (pawanpinjarkar) [reply pkg/asset/agent/installconfig.go:20]: Can not use it from the imported package as that const is not exported in that package. I do not want to touch that package as it's not owned by us and will need approvals from the installer team which might delay merging this PR.

File: pkg/asset/agent/installconfig_test.go
Comment (pawanpinjarkar) [reply pkg/asset/agent/installconfig_test.go:172]: same as above

File: pkg/asset/agent/installconfig.go
Comment (bfournie) [reply pkg/asset/agent/installconfig.go:20]: OK sure.

File: pkg/asset/agent/installconfig_test.go
Comment (bfournie) [reply pkg/asset/agent/installconfig_test.go:172]: You should be able to use the constant you defined in agent/installconfig.go

File: pkg/asset/agent/installconfig.go
Comment (rwsu) [inline pkg/asset/agent/installconfig.go:132]:
```suggestion
			allErrs = append(allErrs, field.Required(fieldPath, fmt.Sprintf("apiVIP must be set for %s platform", baremetal.Name)))
```

File: pkg/asset/agent/installconfig.go
Comment (rwsu) [inline pkg/asset/agent/installconfig.go:136]:
```suggestion
			allErrs = append(allErrs, field.Required(fieldPath, fmt.Sprintf("ingressVIP must be set for %s platform", baremetal.Name)))
```

File: pkg/asset/agent/installconfig.go
Comment (rwsu) [inline pkg/asset/agent/installconfig.go:143]:
```suggestion
			allErrs = append(allErrs, field.Required(fieldPath, fmt.Sprintf("apiVIP must be set for %s platform", vsphere.Name)))
```

File: pkg/asset/agent/installconfig.go
Comment (rwsu) [inline pkg/asset/agent/installconfig.go:147]:
```suggestion
			allErrs = append(allErrs, field.Required(fieldPath, fmt.Sprintf("ingressVIP must be set for %s platform", vsphere.Name)))
```

File: pkg/asset/agent/installconfig.go
Comment (rwsu) [reply pkg/asset/agent/installconfig.go:157]: A comment here saying SNO = None platform will help future maintainers.

**Review by rwsu — CHANGES_REQUESTED** [review]

File: pkg/asset/agent/installconfig_test.go
Comment (pawanpinjarkar) [reply pkg/asset/agent/installconfig_test.go:172]: yes :slightly_smiling_face:

**Comment by lranjbar** [comment]
/approve

Worked for me when I tested it locally.

**Pull request merged**
